http/cookie.ts

```diff
@@ -1,11 +1,103 @@
 // Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
-import { ServerRequest } from "./server.ts";
+import { ServerRequest, Response } from "./server.ts";
+import { assert } from "../testing/asserts.ts";
+import { pad } from "../strings/pad.ts";
 
 export interface Cookie {
   [key: string]: string;
 }
 
-/* Parse the cookie of the Server Request */
+export interface CookieValue {
+  name: string;
+  value: string;
+}
+
+export interface CookieOptions {
+  Expires?: Date;
+  MaxAge?: number;
+  Domain?: string;
+  Path?: string;
+  Secure?: boolean;
+  HttpOnly?: boolean;
+  SameSite?: SameSite;
+}
+
+export type SameSite = "Strict" | "Lax";
+
+export function cookieDateFormat(date: Date): string {
```

Comment:
Is this a specific date format for cookies? Does it have a name? If it's not just for cookies, consider moving this to //datetime

Reply:
I was thinking it was specific but no. It's called `IMF-fixdate`. I'll move it to datetime. Also is `date.toIMF()` good naming for you?

Reply:
Is there a name for this in Go? Perhaps you can borrow that.

Reply:
No, the only mention of it i can find is IMF-fixdate

Reply:
I think you can call it "RFC1123_GMT"

We don't have the same way of serializing datetime formats as Go. I'd like to slowly move in that direction tho. So I think you can just keep the `toIMF()` where it is now. But please  familiarize yourself with how I think this would be ideally structured:
https://github.com/golang/go/blob/f67f5511ee0f225bcc8943994ba6139eed375e85/src/net/http/server.go#L903-L909
https://github.com/golang/go/blob/f67f5511ee0f225bcc8943994ba6139eed375e85/src/net/http/cookie.go#L200-L204
Consider adding some of the commentary from Go to the jsdoc of "toIMF()"